src/java.base/share/classes/java/lang/constant/PackageDesc.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.lang.constant;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A nominal descriptor for a {@code Package} constant {@jvms 4.4.12}.
+ *
+ * <p>To create a {@linkplain PackageDesc} for a package, use {@link #of} or
+ * {@link #ofInternalName(String)}.
+ *
+ * @since 21
+ */
+public sealed interface PackageDesc
+        permits PackageDescImpl {
+
+    /**
+     * Returns a {@linkplain PackageDesc} for a package,
+     * given the name of the package, such as {@code "java.lang"}.
+     * <p>
+     * {@jls 13.1}
+     *
+     * @param name the fully qualified (dot-separated) binary package name
```

Comment:
JLS 13.1 does not describe a package has a binary name.   Is it correct to say "binary package name"?

```suggestion
     * @param name the fully qualified (dot-separated) package name
```